system/helper/functions.php

```diff
@@ -327,15 +327,22 @@ function deserialize($varValue, $blnForceArray=false)
 		return $blnForceArray ? array() : '';
 	}
 
-	// Potentially including an object (see #6724)
-	if (preg_match('/[OoC]:\+?[0-9]+:"/', $varValue))
+	if (version_compare(PHP_VERSION, '7.0', '>='))
 	{
-		trigger_error('The deserialize() function does not allow serialized objects', E_USER_WARNING);
-
-		return $blnForceArray ? array($varValue) : $varValue;
+		$varUnserialized = @unserialize($varValue, array('allowed_classes' => false));
```

Comment:
I don’t think so. The unserialized result would just have `__PHP_Incomplete_Class` objects in it. My test didn’t show warning messages:

```php
unserialize('O:8:"stdClass":1:{s:3:"foo";s:3:"bar";}', ['allowed_classes' => false]);
/* Results in: 
__PHP_Incomplete_Class Object (
    [__PHP_Incomplete_Class_Name] => stdClass
    [foo] => bar
) 
*/
```